File: replaceWithAlphabet/kata.py

chars = 'abcdefghijklmnopqrstuvwxyz'
char_numbers = {chars[i] : i+1 for i in range(26)} # create number of characters

def alphabet_position(text: str):
    # print(char_numbers)
    output = ''
    for char in text.lower():
        try: 
            output += str(char_numbers[char]) + ' '
            # print(char)
            # print(char_numbers[char])
        except :
            continue

    
    return output[:-1]



print(alphabet_position("The sunset sets at twelve o' clock."))

File: replaceWithAlphabet/kata.js

chars = 'abcdefghijklmnopqrstuvwxyz';
charNumbers = {}
function alphabetPosition(text) {
    // Fill charNumbers
    let i = 1
    for (const char of chars) {
        charNumbers[char] = String(i)
        i++;
    }

    // create output
    let output = ''
    for (const char of text.toLowerCase()) {
        // console.log(char);
        // console.log(charNumbers[char]);
        if (charNumbers[char] === undefined) continue;
        output += charNumbers[char] + ' ';
    }
    // console.table(charNumbers)

  return output;
}


console.log(alphabetPosition("The sunset sets at twelve o' clock."));